<?xml version="1.0"?>
<launch>
  <arg name="tfpose_output_topic" value="tfpose_ros/pose" />
  <arg name="camera_name" value="myxtion" />
  <arg name="skeleton_topic" value="skeleton_to_3d/skeleton" />
  <arg name="sensor_namespace" value="sensor1" />
  
  <include file="$(find skeleton3d)/launch/generate_skeleton3d.launch">
    <arg name="pointcloud" value="$(arg camera_name)/depth_registered/points" />
    <arg name="sensor_namespace" value="$(arg sensor_namespace)" />
    <arg name="skeleton_topic" value="$(arg skeleton_topic)" />
    <arg name="camera_name" value="$(arg camera_name)" />
  </include>
  
  <node name="skeleton_to_3d_vis" pkg="skeleton3d" type="skeleton_visualiser" output="screen">
    <param name="input_skeleton" type="str" value="/$(arg sensor_namespace)/$(arg skeleton_topic)" />
    <param name = "output_marker" type="str" value="/skeleton_to_3d_vis/skeleton_vis" />
    <param name="camera_name" type="str" value="$(arg camera_name)" />
    <param name="decaying_lines" type="bool" value="false" />
  </node>

  <node name="pose_estimator_vis" pkg="tfpose_ros" type="visualization.py" output="screen" ns="$(arg sensor_namespace)" >
    <param name="camera" value="/$(arg sensor_namespace)/$(arg camera_name)/rgb/image_raw" />
    <param name="pose" value="/$(arg sensor_namespace)/pose_estimator/pose" />
  </node>


  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find skeleton3d)/rviz/playbag_markers.rviz" />
</launch>
